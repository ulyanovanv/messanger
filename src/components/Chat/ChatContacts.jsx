import React, { Fragment } from 'react';
import PropTypes from 'prop-types';

import images from '../../helpers/images.js';

export default function ChatContacts(props) {
  const { contacts } = props;

  const chatContacts = contacts.map(obj => (
    <Fragment key={obj.name}>
      <img
        src={images[obj.image]}
        className="mx-2"
        title={obj.image}
        alt={obj.image}
      />
      <h6>{obj.name}</h6>
    </Fragment>
  ));

  return (
    <div className="App__chats_contact d-flex flex-row justify-content-center align-items-end border-bottom py-2">
      {chatContacts}
    </div>
  );
}

ChatContacts.propTypes = {
  contacts: PropTypes.arrayOf(PropTypes.object),
};
