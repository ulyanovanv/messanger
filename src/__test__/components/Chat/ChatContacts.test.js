import React from 'react';
import ChatContacts from './../../../components/Chat/ChatContacts.jsx';
import renderer from 'react-test-renderer';

test('ChatContacts', () => {
  let contacts = [
    {
      name: "Ara Targaryen",
      image: "ara"
    },
    {
      name: "Kakadu Lannister",
      image: "kakadu"
    }
  ];

  let tree = renderer.create(<ChatContacts contacts={contacts}/>).toJSON();
  expect(tree).toMatchSnapshot();
});
