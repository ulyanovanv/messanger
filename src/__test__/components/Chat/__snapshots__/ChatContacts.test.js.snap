// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatContacts 1`] = `
<div
  className="App__chats_contact d-flex flex-row justify-content-center align-items-end border-bottom py-2"
>
  <img
    className="mx-2"
    src="test-file-stub"
    title="ara"
  />
  <h6>
    Ara Targaryen
  </h6>
  <img
    className="mx-2"
    src="test-file-stub"
    title="kakadu"
  />
  <h6>
    Kakadu Lannister
  </h6>
</div>
`;
